transitmatters/new-train-tracker
# fix(predictions): only fetch predictions for a train when user opens the train opover

## Motivation
Currently, we fetch departure predictions for every train every time the page is loaded. This is making the NTT slow, and leading to our hitting V3 API limits.



## Changes
Only call the prediction API when the user opens the popover for a train

Another way to do this might be to call for predictions on a per-line basis, store them in the `useMbtaApi` hook, and then match them up with vehicles. However, this would mean the predictions only get updated whenever the train positions are updated. 



## Testing Instructions
Run the app locally and verify that the prediction API is not called until you open the pop-over for a vehicle.
